File: src/frontend/src/apis/services/achievement.service.ts
import { publicClient } from '../client'
import { AchievementDTO, AchievementStatsDTO, PageDTO, ResponseMessage } from '../schema/type'

export interface AchievementQueryParams {
  type?: string
  awardLevel?: string
  year?: number
  page?: number
  size?: number
}

export const AchievementService = {
  /**
   * 获取成就列表（分页）
   */
  getAchievements: async (params?: AchievementQueryParams) => {
    const response = await publicClient.get<ResponseMessage<PageDTO<AchievementDTO>>>(
      '/achievements',
      {
        params,
      }
    )
    return response.data
  },

  /**
   * 获取成就统计
   */
  getAchievementStats: async () => {
    const response =
      await publicClient.get<ResponseMessage<AchievementStatsDTO>>('/achievements/stats')
    return response.data
  },
}

File: src/backend/src/main/java/com/bluenet/web/application/service/AchievementService.java
package com.bluenet.web.application.service;

import com.bluenet.web.api.dto.PageDTO;
import com.bluenet.web.api.dto.achievement.AchievementDTO;
import com.bluenet.web.api.dto.achievement.AchievementStatsDTO;
import com.bluenet.web.api.dto.achievement.CreateAchievementRequestDTO;
import com.bluenet.web.api.dto.achievement.UpdateAchievementRequestDTO;
import com.bluenet.web.domain.model.enumerate.AchievementType;
import com.bluenet.web.domain.model.enumerate.AwardLevel;

public interface AchievementService {
    PageDTO<AchievementDTO> getAchievements(Integer page, Integer size, AchievementType type, AwardLevel awardLevel,
            Integer year);

    AchievementStatsDTO getAchievementStats();

    AchievementDTO createAchievement(CreateAchievementRequestDTO request);

    AchievementDTO updateAchievement(Long id, UpdateAchievementRequestDTO request);

    void deleteAchievement(Long id);
}
